src/main/java/teammates/ui/newcontroller/GetStudentCoursesAction.java

```diff
@@ -0,0 +1,175 @@
+package teammates.ui.newcontroller;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.http.HttpStatus;
+
+import teammates.common.datatransfer.CourseDetailsBundle;
+import teammates.common.datatransfer.FeedbackSessionDetailsBundle;
+import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.util.Const;
+
+/**
+ * Action: gets the courses and feedback sessions which a student is enrolled in.
+ */
+public class GetStudentCoursesAction extends Action {
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        // Unregistered users may also be able to access
```

Comment:
This shouldn't be the case. Student home page is accessible only to registered students.

Reply:
That is because the previous `StudentHomePageAction` does not redirect to the unauthorized page if the user is not registered; rather it will show the "Oops! Your account is not known ..." status message. It is a hack that works for SSR, but not for REST.